crates/c-api/include/wasmtime/component/instance.h

```diff
@@ -0,0 +1,27 @@
+#ifndef WASMTIME_COMPONENT_INSTANCE_H
+#define WASMTIME_COMPONENT_INSTANCE_H
+
+#ifdef WASMTIME_FEATURE_COMPONENT_MODEL
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct wasmtime_component_instance_t wasmtime_component_instance_t;
```

Comment:
I think we'll want this to be similar to `wasmtime_instance_t` where it's not an opaque struct but instead it's a known sturct of two integers (although embeddings won't use the integers much). That'll also obviate the need for deletion of this structure too